src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -600,11 +600,12 @@ def _infer_schema_for_file_format(
                 drop_tmp_file_format_if_exists_query = (
                     drop_file_format_if_exists_statement(file_format_name)
                 )
-            results = self._session._conn.run_query(infer_schema_query)["data"]
-            if len(results) == 0:
+            stage_files = self._session._conn.run_query(f"LIST '{path}'")["data"]
+            if len(stage_files) == 0:
                 raise FileNotFoundError(
                     f"Given path: '{path}' could not be found or is empty."
                 )
```

Comment:
will empty files still flag this error?

Reply:
It should not, this piece of code check whether given file exist and raise exception when it does not exist, empty file will not trigger it

Reply:
in that case, we should remove `"or is empty."` from error message.

Reply:
you are right! removed